backend/src/yeomanui.ts

```diff
@@ -137,27 +136,13 @@ export class YeomanUI {
       const meta: Environment.GeneratorMeta = this.getGenMetadata(generatorName);
       // TODO: support sub-generators
       env.register(meta.resolved);
-      const gen: any = env.create(meta.namespace, {});
-      // check if generator defined a helper function called getPrompts()
-      const genGetPrompts = _.get(gen, "getPrompts");
-      if (genGetPrompts) {
-        const promptNames: any[] = genGetPrompts();
-        const prompts: IPrompt[] = promptNames.map(value => {
-          return _.assign({ questions: [], name: "", description: "" }, value);
-        });
-        this.setPrompts(prompts);
-      }
 
-      const genGetImage = _.get(gen, "getImage");
-      if (genGetImage) {
-        const image: any = genGetImage();
-        if (image.then) {
-          image.then((contents: string) => {
-            this.logger.debug(`image contents: ${contents}`);
-          });
-        } else if (image !== undefined) {
-          this.logger.debug(`image contents: ${image}`);
-        }
+      const getGenMNamespace = this.getGenNamespace(generatorName);
+      const gen: any = env.create(getGenMNamespace, {});
+      // check if generator defined a helper function called setPromptsCallback()
+      const setPromptsCallback = _.get(gen, "setPromptsCallback");
+      if (setPromptsCallback) {
+        setPromptsCallback(this.setPromptList.bind(this));
```

Comment:
Is it safe to merge this PR before repleacing old generator implementation with setPromptsCallback ?

Reply:
Yes